src/bin/pg_autoctl/pgsql.c

```diff
@@ -637,6 +648,350 @@ pgsql_drop_replication_slot(PGSQL *pgsql, const char *slotName, bool verbose)
 }
 
 
+/*
+ * pgsql_drop_replication_slots drops all the replication slots. If the verbose
+ * flag is false, then no info message will be logged.
+ */
+bool
+pgsql_drop_replication_slots(PGSQL *pgsql, bool verbose)
+{
+	char *sql =
+		"SELECT pg_drop_replication_slot(slot_name) "
+		"  FROM pg_replication_slots ";
```

Comment:
Might be useful to only drop 'streaming' slots and not the logical ones.

Reply:
And we have a pattern for the replication slot names that pg_auto_failover should care about, too. So now we're using it to drop only “our” slots. Thanks.